cli/main.c

```diff
@@ -1497,7 +1497,8 @@ static int fw_read(int argc, char **argv)
 	}
 
 	progress_start();
-	ret = switchtec_fw_read_fd(cfg.dev, cfg.out_fd, inf->part_addr,
+	ret = switchtec_fw_read_fd(cfg.dev, cfg.out_fd,
+				   inf->part_addr + inf->part_body_offset,
```

Comment:
I'd suggest it would be better to pass 'inf' directly to switchtec_fw_read_fd() instead of requiring the caller to know the exact formula for the arguments.

Reply:
I suppose this API is to read at any offset of any bytes from a partition, so I think we'd better leave the flexibility to user rather than forcing it to read from the beginning of the image body. Or we can provide another API basing on this just to read the body as a counterpart of the write_hdr().

Reply:
Well it's questionable why anyone would want to read the firmware partially. So, I'd probably just say we don't need it (until someone actually does have a use-case).

But yes, if there's another API function that just takes the inf struct and applies the necessary math that would be ok too.